packages/catalog-model/src/entity/policies/DefaultParentEntityPolicy.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '../Entity';
+import { GroupEntity } from '../../kinds';
+import { EntityPolicy } from './types';
+import { DEFAULT_NAMESPACE } from '../constants';
+import { parseEntityRef, stringifyEntityRef } from '../ref';
+
+/**
+ * DefaultParentPolicy is an EntityPolicy that updates group entities
+ * with a parent of last resort. This ensures that, while we preserve
+ * any existing group hierarchies, we can guarantee that there is a
+ * single global root of the group hierarchy.
+ *
+ * @public
+ */
+export class DefaultParentEntityPolicy implements EntityPolicy {
```

Comment:
Maybe change the name to `GroupDefaultParentEntityPolicy`.

Reply:
Happy to change this, but I named it to be consistent with `DefaultNamespaceEntityPolicy`.

Reply:
that one is for any entity, while this one is only for groups - could be worth adding that word in there